### PR TITLE
Refactoring and small improvements

### PR DESCRIPTION
**Pull request by anjensan**
PR #209 depends on this PR.

### CODE REVIEW

**Comment by iwko** [comment]
Which task is connected to this pull-request?

**Comment by anjensan** [comment]
> Which task is connected to this pull-request?

This is an outcome of "[opportunistic refactoring](https://martinfowler.com/bliki/OpportunisticRefactoring.html)" from #209

File: bigflow/build/pip.py [hunk 1]
```diff
@@ -79,6 +100,25 @@ def maybe_recompile_requirements_file(req_txt: Path) -> bool:
...
+    logger.debug("Scan all req.in-like files: %s", req_in)
+    c = req_in.read_text()
+    yield c
+    for include in re.findall(r"\s*-r\s+(.*)", c):
+        include: str
+        fn = include.split("#", 1)[0].strip()
+        yield from _collect_all_input_files_content(req_in.parent / fn)
+
+
+def reqin_file_hash(req_in: Path):
```
Comment (beczkowb) [inline bigflow/build/pip.py:113]: use full names everywhere please, f.e. reqin -> requirements_in

File: bigflow/commons.py [hunk 2]
```diff
@@ -27,34 +28,78 @@ def now(template: str = "%Y-%m-%d %H:00:00"):
     return datetime.now().strftime(template)
 
 
+class _OutDumper(threading.Thread):
```
Comment (beczkowb) [inline bigflow/commons.py:31]: Maybe `_StreamOutputDumper`?

File: test/mixins.py [hunk 3]
```diff
@@ -13,7 +13,39 @@
 from pathlib import Path
 
 
-class TempCwdMixin(unittest.TestCase):
+class Mixin(unittest.TestCase):
```
Comment (beczkowb) [inline test/mixins.py:16]: Do we need this class? Can't you just do `FileUtilsMixin(unittest.TestCase)`?

**Review by beczkowb — CHANGES_REQUESTED** [review]
needs work

File: bigflow/build/pip.py (hunk 1 above)
Comment (anjensan) [reply bigflow/build/pip.py]: done

File: bigflow/commons.py (hunk 2 above)
Comment (anjensan) [reply bigflow/commons.py]: done

File: test/mixins.py (hunk 3 above)
Comment (anjensan) [reply test/mixins.py:16]: Well, strictly it is not required. We even can drop inheritance from `unittest.TestCase` completely (but user/client need to remember to put `TestCase` as the last class in the inheritance list). New class `Mixin` is added as "self-documented" code (and also eventually some mro-check may be added here).

**Review by beczkowb — APPROVED** [review]

**Pull request merged**